DPGAnalysis/MuonSysAging/plugins/CSCChamberMasker.cc

```diff
@@ -0,0 +1,333 @@
+
+}
+
+
+CSCChamberMasker::~CSCChamberMasker()
+{
+ 
+}
+
+
+//
+// member functions
+//
+
+// ------------ method called to produce the data  ------------
+void
+CSCChamberMasker::produce(edm::Event& event, const edm::EventSetup& conditions)
+{
+  
+  edm::Service<edm::RandomNumberGenerator> randGenService;
+  CLHEP::HepRandomEngine& randGen = randGenService->getEngine(event.streamID());
+ 
+  std::unique_ptr<CSCStripDigiCollection> filteredStripDigis(new CSCStripDigiCollection());
+  std::unique_ptr<CSCWireDigiCollection> filteredWireDigis(new CSCWireDigiCollection());
+  std::unique_ptr<CSCCLCTDigiCollection> filteredCLCTDigis(new CSCCLCTDigiCollection());
+  std::unique_ptr<CSCALCTDigiCollection> filteredALCTDigis(new CSCALCTDigiCollection());
+
+  // Handle strip digis
+  if(!m_stripDigiToken.isUninitialized())
+  {
+      edm::Handle<CSCStripDigiCollection> cscStripDigis;
+      event.getByToken(m_stripDigiToken, cscStripDigis);
+
+      for (CSCStripDigiCollection::DigiRangeIterator j=cscStripDigis->begin(); j!=cscStripDigis->end(); j++) {
+          std::vector<CSCStripDigi>::const_iterator digiItr = (*j).second.first;
+          std::vector<CSCStripDigi>::const_iterator last = (*j).second.second;
+
+          CSCDetId const cscDetId=(*j).first;
+
+          // Since lookups are chamber-centric, make new DetId with layer=0
+          CSCDetId chId = CSCDetId(cscDetId.endcap(), cscDetId.station(), cscDetId.ring(), cscDetId.chamber(), 0);
+          // std::cout<<"Det id: " << chId<<std::endl;
+
+          for( ; digiItr != last; ++digiItr) {
+
+              auto chEffIt = m_CSCEffs.find(chId);
+
+              if (chEffIt != m_CSCEffs.end()) {
+                  // std::cout<<"  Strip Digi in chamber of interest: " << chId <<std::endl;
+                  std::pair<unsigned int, float> typeEff = chEffIt->second;
+                  int type = typeEff.first % 10; // second digit gives type of inefficiency
+                  int layer = typeEff.first/10; // first digit gives layer (0 = chamber level)
+                  // std::cout << " type: " << type << " layer: " << layer << std::endl;
+                  bool doRandomize = false;
+                  // if ((type == EFF_STRIPS || type == EFF_CHAMBER) && (layer == 0 || cscDetId.layer() == layer) && randGen.flat() <= typeEff.second) {
+                  if ((type == EFF_STRIPS || type == EFF_CHAMBER) && (layer == 0 || cscDetId.layer() == layer)) doRandomize = true;
+
+                  if (!doRandomize  || (randGen.flat() <= typeEff.second)) {
+                      // std::cout << "We're keeping this strip digi in " << cscDetId << std::endl;
+                      filteredStripDigis->insertDigi(cscDetId,*digiItr);
+                  } else {
+                      // std::cout << "We're !!dropping!! this digi in " << cscDetId << std::endl;
+                  }
+              } 
+          }
+      }
+  }
+
+  // Handle wire digis
+  if(!m_wireDigiToken.isUninitialized())
```

Comment:
This loop looks exactly the same as the previous loop, just over a different collection. If so, better to use a function template to reduce code duplication.